frontend/src/metabase-lib/lib/queries/NativeQuery.ts

```diff
@@ -41,31 +45,110 @@ export const NATIVE_QUERY_TEMPLATE: NativeDatasetQuery = {
     "template-tags": {},
   },
 };
-// This regex needs to match logic in replaceCardId and _getUpdatedTemplateTags.
-const CARD_TAG_REGEX = /^#([0-9]*)$/;
 
-function cardTagCardId(name) {
-  const match = name.match(CARD_TAG_REGEX);
+///////////////////////////
```

Comment:
This new version is beautiful.

I may have expressed myself badly in today's comment, because imo this new function should live in that extracted file. Not in this class file.

If you're done tweaking this, I'll understand.